packages/components/src/Dropdown/Dropdown.tsx

```diff
@@ -287,8 +401,19 @@ const DropdownOptions = function (
   props: PropsWithRenderProp<{ open: boolean }>,
 ) {
   const { className, ...rest } = props;
+  const { compact, optionsAlign, optionsClassName } =
+    useContext(DropdownContext);
+
   return (
-    <Listbox.Options className={clsx(styles.options, className)} {...rest} />
+    <Listbox.Options
+      className={clsx(
+        styles.options,
+        className,
+        optionsAlign === "right" && styles.optionsAlignRight,
+        optionsClassName || (!compact && styles.optionsFullWidth),
```

Comment:
Mm, I think it's a bit odd that if you pass in an `optionsClassName` and you're not using the `compact` variant, the width will suddenly be off because the full width styling was removed. Maybe the concatenation method would work better here where `optionsClassName`, if it exists, is added to `styles.optionsFullWidth` if it's not in compact mode.

Reply:
Thanks for the comment @dierat . This is why I started with `optionsWidth` to distinguish between either user defined width or full width. If I concatenate, then the full width will take precedence when dev supplied optionsClassName with inline Tailwind (ex: `optionsClassName="w-96"`). Even if dev defined the style in a module, it is unclear which should take precedence, unless dev adds `!important`. I think this ambiguity is not that great. That said, I totally agree with you that it is somewhat opaque that providing `optionsClassName` will remove full width. But I suppose 1. I can add more documentation and 2. it is more visually evident no width is provided and that dev needs to add width to `optionsClassName`. What do you think @dierat @anniehu4 ?

Reply:
Ah, I see. Maybe it would be enough to add a comment on the `optionsClassName` prop on line 72 explaining that if they're not using the `compact` variant, they will need to explicitly set the width in those styles?

Reply:
Sounds good. I've added documentation on line 72.